src/screens/ListScreen.js

```diff
@@ -17,37 +18,125 @@ const styles = StyleSheet.create({
+};
+
+const SearchCardsComp = ({
+  scroll,
+  navigation,
+  marginBottom,
+  label,
+  labelAlign,
+  addHistory,
+  searchStack,
+}) => (
+  <View>
+    <Inset>
+      <Text
+        style={{
+          fontSize: 15,
+          color: '#3b3f51',
+          marginTop: 10,
+          textAlign: labelAlign,
+        }}
+      >
+        {label}
+      </Text>
+    </Inset>
+    <View style={{ marginTop: 10, marginBottom }}>
+      {scroll.map(tree => {
+        const { uuid, name, comments } = tree;
+        return (
+          <SearchCard
+            key={uuid}
+            name={name}
+            comments={comments}
+            onPress={() => {
+              navigation.push('TreeDetails', { uuid });
+              if (addHistory) {
+                if (searchStack.length < 1) {
+                  searchStack.unshift(tree);
+                } else if (tree.name !== searchStack[0].name) {
+                  searchStack.unshift(tree);
+                }
+                if (searchStack.length > 5) {
+                  searchStack.pop();
+                }
+                storeSearchStack(searchStack);
+              }
+            }}
+          />
+        );
+      })}
+    </View>
+  </View>
+);
+
+SearchCardsComp.propTypes = {
+  scroll: shape({
+    push: func,
+  }),
+  navigation: shape({
+    push: func,
+  }),
+  marginBottom: number,
+  label: string,
+  labelAlign: string,
+  addHistory: bool,
+  // eslint-disable-next-line react/forbid-prop-types
+  searchStack: array,
+};
+
+export default function ListScreen({
+  navigation,
+  style,
+  data,
+  searchStack,
+  searchQuery,
+  searchEntered,
+}) {
+  let text;
+  if (searchQuery.length === 0) {
+    text = 'All trees';
+  } else if (searchEntered === true) {
+    text = `${data.length} ${data.length === 1 ? 'result' : 'results'}`;
+  } else if (data.length === 0) {
+    text = "Oops, that's an invalid search. Try again!";
+  } else {
+    text = 'Suggestions';
+  }
   return (
     <ScrollView style={[styles.list, style]}>
-      <Inset>
-        <Text
-          style={{
-            fontSize: 11,
-            color: '#3b3f51',
-            marginTop: 48,
-            textAlign: 'right',
-          }}
-        >
-          {`${data.length} ${data.length === 1 ? 'result' : 'results'}`}
-        </Text>
-      </Inset>
-      <View style={{ marginTop: 10, marginBottom: 248 }}>
-        {data.map(tree => {
-          const { uuid, name, comments } = tree;
-          return (
-            <SearchCard
-              key={uuid}
-              name={name}
-              comments={comments}
-              onPress={() => navigation.push('TreeDetails', { uuid })}
-            />
-          );
-        })}
+      <View>
+        {searchQuery.length === 0 && searchStack.length !== 0 ? (
+          <SearchCardsComp
+            scroll={searchStack}
+            addHistory={false}
+            navigation={navigation}
+            marginBottom={0}
+            label="Recents"
```

Comment:
they're able to scroll past the recently seen searches right -- does the label update?